mirage/mimic_happy_eyeballs.mli

```diff
@@ -25,11 +30,7 @@ end
     user. *)
 module Make
     (Stack : Tcpip.Stack.V4V6)
-    (DNS : Dns_client_mirage.S with type Transport.stack = Stack.t)
```

Comment:
while this compiles, and is not strictly necessary -- I currently think it would be beneficial if we'd get the DNS back here -- since then the passed Happy_eyeballs would be ensured to have the DNS resolving power.